js/dashboard.js

```diff
@@ -252,4 +790,147 @@
 		// Initial update.
 		updateDashboard(false);
 	}
+
+	/**
+	 * Add a CSV export button to a table.
+	 *
+	 * @param {HTMLTableElement} table Table to process
+	 */
+	function addExportButton(table) {
+		const exportBtn = document.createElement('a');
+		exportBtn.classList.add('button');
+		exportBtn.role = 'button';
```

Comment:
As the `href` attribute is used here to trigger the download, in my opinion it’s a link and not a button, so shouldn’t get the `button` role (and if it would be a button semantically, should use the `button` element).

The problem with setting the `href` on click is, that a link is not focusable via e.g. keyboard without an `href`. Would it be possible to populate the `href` on load?